.asf.yaml

```diff
@@ -31,6 +33,23 @@ github:
     squash:  true
     merge:   true
     rebase:  true
+  protected_branches:
+    # no force push to master
+    master:
+#      required_pull_request_reviews:
+#        required_approving_review_count: 1
+#        dismiss_stale_reviews: false
+      required_status_checks:
+          # strict means "Require branches to be up to date before merging".
+          strict: false
+          # list of names of checks that must pass
+#          contexts:
+#            - Commit Validation on Linux/JDK 11
```

Comment:
we could probably set up a meta-job which serves as check for all other jobs, if we decide to turn this on.

Reply:
@neilcsmith-net status checks look at the `contexts` list and require all checks of that particular name to be green. We would need some meta-job for this to handle edge cases like `ci:no-build`. Otherwise we could have used "Cleanup Workflow Artifacts". Since if that one is green, everything is fine.

Reply:
Ah, OK, your meta-job comment now makes sense to me!  Mind you, we should possibly do some basic paperwork checks (eg. commit info) for all merged PRs? `ci:no-build` could mean skip the build and dependent checks rather than no actions at all?

Reply:
the second half of the paperwork job requires the build artifact for verify-libs-and-licenses, check-sigtests-release and check-sigtests-release. So we would have to split it to have one part of it before/without the build and the other as normal secondary, post-build job.

Although I am wondering what happens if B depends on A, A is skipped and B is set to `always()`. If it still runs we might have our way out without having to split the job. Something I can test in my sandbox.

As a workaround we could simply avoid `ci:no-build` on PRs from new contributors so that we have the option to inspect the papers. (I mean inspection via checkout is always possible, thats how I used to do it)

Reply:
> PRs from new contributors so that we have the option to inspect the papers.

That suggests it's new contributors rather than existing committers / PMC members whose eg. invalid noreply email addresses in commits needs to be flagged up! :wink:

Reply:
I am going to try to make this work somehow. (but yes I added it primarily to check new contributors, I didn't check committer/PMC emails during reviews so far).
The "check" doesn't do anything btw. All it does is to print the details of the last 10 commits.